### DISCUSSION TITLE
[react-markdown] Cannot compile with webpack 5

### DISCUSSION THREAD

**Discussion by nikhilDigiplusIT** [remarkjs/remark#903] [Q&A] [answered]
Affected packages and versions
7.1.0

Steps to reproduce
Set webpack configuration as [here](https://github.com/vfile/vfile/issues/38#issuecomment-640479137).

Build with webpack 5.

Expected behavior
Compiled without errors.

Actual behavior
Got errors below:

```
ERROR in ./node_modules/uvu/node_modules/kleur/index.mjs 4:11-18
Module not found: Error: Can't resolve 'process/browser' in '/Users/sowicm/repos/monoweb/packages/web/node_modules/uvu/node_modules/kleur'
Did you mean 'browser.js'?
BREAKING CHANGE: The request 'process/browser' failed to resolve only because it was resolved as fully specified
(probably because the origin is a '*.mjs' file or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.
```

**Solutions that I tried which didn't work**
1. Added process/browser to the webpack plugins
` new webpack.ProvidePlugin({
             process: 'process/browser',
      })`

**Comment by ChristianMurphy** [comment, marked as answer]
@nikhilDigiplusIT did you try the suggestion noted in the error message?

> BREAKING CHANGE: The request 'process/browser' failed to resolve only because it was resolved as fully specified
> (probably because the origin is a '*.mjs' file or a '*.js' file where the package.json contains '"type": "module"').
> The extension in the request is mandatory for it to be fully specified.
> Add the extension to the request.

Implies that changing

```diff
- new webpack.ProvidePlugin({ process: 'process/browser', })
+ new webpack.ProvidePlugin({ process: 'process/browser.js', })
```

may work

**Reply by nikhilDigiplusIT** [reply]
It didn't work.

**Reply by wooorm** [reply]
My friend. You’re commenting on open source software. This is not customer service you’re entitled to. You’re not paying anyone.

Ask proper questions and maybe you’ll get help: https://github.com/remarkjs/.github/blob/main/support.md.

**Reply by charliematters** [reply]
I've run into the same problem and adding the `.js` has resolved the webpack build error for me I'm afraid.
react-markdown 7.1.0
webpack 5.48.0

**Reply by wooorm** [reply]
please provide more information, this is not enough to go on. There are many users using webpack with react-markdown fine.

**Reply by charliematters** [reply]
What more information do you require? I am using webpack and it is now working as a result of your suggestion

**Reply by charliematters** [reply]
To be clear, I wanted anyone coming across this issue to see that it is a valid solution (in my case at least) instead of being put off by seeing it not work for someone.

**Reply by wooorm** [reply]
Oh it did work, it sounded like it didn’t. Sorry!

glad it works :)

**Reply by wooorm** [reply]
(hiding the above misunderstanding to help future folks find the solution!)

**Reply by keslo** [reply]
It's work! Thanks

**Comment by ashleydavis** [comment]
This is still an issue for me.

I've prepared a simplified version of my setup to show the issue:

https://github.com/ashleydavis/react-markdown-with-webpack-5

I've tried both `new webpack.ProvidePlugin({ process: 'process/browser', })` and `new webpack.ProvidePlugin({ process: 'process/browser.js', })`.

**Comment by ashleydavis** [comment]
I'm using the latest React-markdown and Webpack as of today.

**Comment by ashleydavis** [comment]
This is the result I'm getting in the browser:

![image](https://user-images.githubusercontent.com/625126/168004147-03eddf7b-c725-4406-bb5a-8f0eb4b61920.png)

Can anyone else confirm if this is still a problem for them?

Here's the example, just clone and run `npm install` and `npm start` and you should see the problem:

https://github.com/ashleydavis/react-markdown-with-webpack-5

I'm using Node.js v16.14.0

**Comment by whinc** [comment]
I have the same problem and can't sovle it acoording to the solution above. I have to rollback to version 6.x  to  solve the problem.

**Reply by ashleydavis** [reply]
There is an example here that shows how to use React-markdown v8.0.3 with Webpack 5:
https://github.com/ashleydavis/react-markdown-with-webpack-5

**Reply by ashleydavis** [reply]
Did you try my example code:
https://github.com/ashleydavis/react-markdown-with-webpack-5

Does that work for you?

**Reply by whinc** [reply]
@ashleydavis 
I use [icejs](https://github.com/alibaba/ice) as build tool and cannot apply your solution. I sovle this problem finally with a stupid way on react-markdown@^8.

1. first install polyfill 'process'
```
npm i process
```
2. modify node_modules: add one line to the top of the problem file `node_modules/uvu/node_modules/kleur/index.mjs`
![image](https://user-images.githubusercontent.com/5096493/184793270-7c70ccf0-6983-4109-b8c6-0409d71adb57.png)

compile pass!

**Comment by mizi-lin** [comment]
webpack 5 add rule

```typescript
 {
            test: /\.mjs$/,
            include: /node_modules/,
            type: 'javascript/auto',
            resolve: {
                fullySpecified: false
            }
        }
```
